specification/dartLangSpec.tex

```diff
@@ -2379,28 +2382,40 @@ \subsubsection{The Method \code{noSuchMethod}}
 one of the following is true:
 
 \begin{itemize}
-\item $C$ has a non-trivial \code{noSuchMethod},
-  the interface of $C$ contains $m$,
-  and $C$ has no concrete declaration of $m$
-  (\commentary{that is, no member $m$ is declared or inherited by $C$}).
+\item \textbf{Requested in program:}
+  $C$ has a non-trivial \code{noSuchMethod},
+  the interface of $C$ contains a member signature $S$ named $m$,
```

Comment:
"named $m$": Above $m$ was defined as a *member*, not a name.
So ... "with the same name as $m$"?

Also, we have to define the "NSM forwarded" on *interface members" (signatures), not on class members, because the class does not have that member. That's why we are adding it.

So:

For a concrete class $C$ with and an interface member $m$ of the interface of $C$, $m$ is **NSM-forwarded** if:
* $C$ has a non-trivial NSM,
   $C$ does not have a concrete member which is a valid 
   implementation of $m$, and
   the class $C$ has no declaration of a concrete member implementing $m$,
   and nor does any superclass of $C$.
* or $m$ is declared in a library $L2$ and $C$ is declared in library $L1$ and 
   $m$ is not accessible in $L1$, 
   and neither $C$ nor any superclass of $C$ declares a concrete member
   overriding $m$ (*you can only override something which is accessible 
   at the point of the overriding declaration*).

Reply:
Agreed, we need to consider a member signature in the interface of the class, then the relation to the members of the class (there may or may not be one with the specified name).

I've revised the text, PTAL.

Reply:
Note that the notion of being NSM-forwarded is now a property of a name (which is sufficient, along with accessibility for a specific library if the name is private, to uniquely determine the member signature).

Reply:
Agreed offline.